src/promptflow/promptflow/_core/tool_meta_generator.py

```diff
@@ -238,27 +266,30 @@ def collect_tool_function_in_module(m):
         return tool_methods[0]
 
 
-def generate_python_tool(name, content, source=None):
+def generate_python_tool_meta_dict(name, content, source=None):
     m = load_python_module(content, source)
     f, initialize_inputs = collect_tool_function_in_module(m)
     tool = _parse_tool_from_function(f, initialize_inputs=initialize_inputs)
+    extra_info = getattr(f, "__extra_info")
+    input_settings = getattr(f, "__input_settings")
     tool.module = None
     if name is not None:
         tool.name = name
     if source is None:
         tool.code = content
     else:
         tool.source = source
-    return tool
-
-
-def generate_python_meta_dict(name, content, source=None):
-    return asdict_without_none(generate_python_tool(name, content, source))
+    construct_tool, validate_result = _serialize_tool(tool, input_settings, extra_info)
```

Comment:
Rename `validate_result` to `is_invlid_result`?
(validate is a verb

Reply:
refine variable name